src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7450,9 +7594,17 @@ private static void CombineExtensionMethodArguments(BoundExpression receiver, An
                     CombineExtensionMethodArguments(left, analyzedArguments, actualArguments);
                 }
 
+                CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = binder.GetNewCompoundUseSiteInfo(diagnostics);
+                result = resolveOverloads(binder, methodGroup, actualArguments, isMethodGroupConversion, returnType, returnRefKind, expression, diagnostics, ref useSiteInfo);
+                return true;
+            }
+
+            static MethodGroupResolution resolveOverloads(Binder binder, MethodGroup methodGroup, AnalyzedArguments actualArguments, bool isMethodGroupConversion,
+                TypeSymbol returnType, RefKind returnRefKind, SyntaxNode expression, BindingDiagnosticBag diagnostics, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+            {
                 var overloadResolutionResult = OverloadResolutionResult<MethodSymbol>.GetInstance();
                 bool allowRefOmittedArguments = methodGroup.Receiver.IsExpressionOfComImportType();
-                CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = binder.GetNewCompoundUseSiteInfo(diagnostics);
```

Comment:
>CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = binder.GetNewCompoundUseSiteInfo(diagnostics); [](http://example.com/codeflow?start=16&length=96)

It is rather unconventional to report ```useSiteInfo``` allocated by someone else. I almost made a comment at the call sites that they are failing to report the info. I recommend kipping this line, changing the callers to report their ```useSiteInfo``` prior to calling the method and not passing it.